include/nlohmann/detail/iterators/iter_impl.hpp

```diff
@@ -519,7 +523,11 @@ class iter_impl // NOLINT(cppcoreguidelines-special-member-functions,hicpp-speci
             JSON_THROW(invalid_iterator::create(212, "cannot compare iterators of different containers", m_object));
         }
 
-        JSON_ASSERT(m_object != nullptr);
+        // value-initialized forward iterators can be compared, and must compare equal to other value-initialized iterators of the same type #4493
```

Comment:
Should this comment say something about returning false because neither iterator is less than the other, just so someone reading this doesn't see a mismatch between "must compare equal" and `return false`?

Reply:
Yes, you are right.